Error using in Create-React-App

Hi Thanks for your efforts.
I just created a brand new react project using Create-React-App version `5.0.1`.
Unfortunately after following the doc and scaffolding, when I tried to link my store to the app in my root `index.tsx` file, I get all these errors:

```
Attempted import error: 'print' is not exported from 'graphql' (imported as 'print').
ERROR in ./node_modules/mst-gql/dist/mst-gql.module.js 591:53-58
export 'print' (imported as 'print') was not found in 'graphql' (module has no exports)

ERROR in ./node_modules/mst-gql/node_modules/graphql/index.mjs 25:0-49
Module not found: Error: Can't resolve './version' in '/Users/amin/Code/Edge/web/node_modules/mst-gql/node_modules/graphql'
Did you mean 'version.mjs'?
BREAKING CHANGE: The request './version' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.

ERROR in ./node_modules/mst-gql/node_modules/graphql/index.mjs 27:0-49
Module not found: Error: Can't resolve './graphql' in '/Users/amin/Code/Edge/web/node_modules/mst-gql/node_modules/graphql'
Did you mean 'graphql.mjs'?
BREAKING CHANGE: The request './graphql' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.

ERROR in ./node_modules/mst-gql/node_modules/graphql/index.mjs 29:0-40:50
Module not found: Error: Can't resolve './type' in '/Users/amin/Code/Edge/web/node_modules/mst-gql/node_modules/graphql'
Did you mean 'index.mjs'?
BREAKING CHANGE: The request './type' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.

ERROR in ./node_modules/mst-gql/node_modules/graphql/index.mjs 42:0-48:205
Module not found: Error: Can't resolve './language' in '/Users/amin/Code/Edge/web/node_modules/mst-gql/node_modules/graphql'
Did you mean 'index.mjs'?
BREAKING CHANGE: The request './language' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.

ERROR in ./node_modules/mst-gql/node_modules/graphql/index.mjs 50:0-122
Module not found: Error: Can't resolve './execution' in '/Users/amin/Code/Edge/web/node_modules/mst-gql/node_modules/graphql'
Did you mean 'index.mjs'?
BREAKING CHANGE: The request './execution' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.

ERROR in ./node_modules/mst-gql/node_modules/graphql/index.mjs 51:0-68
Module not found: Error: Can't resolve './subscription' in '/Users/amin/Code/Edge/web/node_modules/mst-gql/node_modules/graphql'
Did you mean 'index.mjs'?
BREAKING CHANGE: The request './subscription' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.

ERROR in ./node_modules/mst-gql/node_modules/graphql/index.mjs 53:0-56:206
Module not found: Error: Can't resolve './validation' in '/Users/amin/Code/Edge/web/node_modules/mst-gql/node_modules/graphql'
Did you mean 'index.mjs'?
BREAKING CHANGE: The request './validation' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.

ERROR in ./node_modules/mst-gql/node_modules/graphql/index.mjs 58:0-91
Module not found: Error: Can't resolve './error' in '/Users/amin/Code/Edge/web/node_modules/mst-gql/node_modules/graphql'
Did you mean 'index.mjs'?
BREAKING CHANGE: The request './error' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.

ERROR in ./node_modules/mst-gql/node_modules/graphql/index.mjs 60:0-97:42
Module not found: Error: Can't resolve './utilities' in '/Users/amin/Code/Edge/web/node_modules/mst-gql/node_modules/graphql'
Did you mean 'index.mjs'?
BREAKING CHANGE: The request './utilities' failed to resolve only because it was resolved as fully specified
(probably because the origin is strict EcmaScript Module, e. g. a module with javascript mimetype, a '*.mjs' file, or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.

webpack compiled with 10 errors
```

Hasn't anyone had this issue with CRA? Apparently this can be fixed with [some change in the webpack config](https://stackoverflow.com/questions/70669718/compile-error-with-a-lot-of-errors-from-webpack) but I don't want to install additional libraries for modifying underlying webpack config and want to stick with CRA for this small project. I hope there is a straightforward solution for this.

--- a/examples/2-scaffolding/package.json
+++ b/examples/2-scaffolding/package.json
@@ -18,7 +18,7 @@
     "@types/react-dom": "^16.8.4",
     "react": "^16.8.6",
     "react-dom": "^16.8.6",
-    "typescript": "^3.4.5"
+    "typescript": "^4.6.4"
   },
   "relativeDependencies": {
     "mst-gql": "../../"

--- a/generator/generate.js
+++ b/generator/generate.js
@@ -1289,9 +1289,6 @@ function scaffold(
 ) {
 
   const schema = graphql.buildSchema(definition)
-  //console.log("+++ schema", definition)
-  // console.log("+++ introspectionQuery", graphql.introspectionQuery, getIntrospectionQuery())
-  //const res = graphql.graphqlSync(schema, getIntrospectionQuery())
   const res = graphql.graphqlSync({schema, source: getIntrospectionQuery()})
   if (!res.data)
     throw new Error("graphql parse error:\n\n" + JSON.stringify(res, null, 2))

--- a/generator/generate.js
+++ b/generator/generate.js
@@ -4,6 +4,7 @@ const graphql = require("graphql")
 const camelcase = require("camelcase")
 const pluralize = require("pluralize")
 const escapeStringRegexp = require("escape-string-regexp")
+const { getIntrospectionQuery } = require("graphql")
 
 const exampleAction = `  .actions(self => ({
     // This is an auto-generated example action.
@@ -1262,8 +1263,12 @@ function scaffold(
     fieldOverrides: []
   }
 ) {
+
   const schema = graphql.buildSchema(definition)
-  const res = graphql.graphqlSync(schema, graphql.introspectionQuery)
+  //console.log("+++ schema", definition)
+  // console.log("+++ introspectionQuery", graphql.introspectionQuery, getIntrospectionQuery())
+  //const res = graphql.graphqlSync(schema, getIntrospectionQuery())
+  const res = graphql.graphqlSync({schema, source: getIntrospectionQuery()})
   if (!res.data)
     throw new Error("graphql parse error:\n\n" + JSON.stringify(res, null, 2))
   return generate(

--- a/generator/mst-gql-scaffold.js
+++ b/generator/mst-gql-scaffold.js
@@ -4,6 +4,7 @@ const path = require("path")
 const fs = require("fs")
 const child_process = require("child_process")
 const graphql = require("graphql")
+const { getIntrospectionQuery } = require("graphql")
 
 const { getConfig, mergeConfigs } = require("./config")
 const { generate, writeFiles, logUnexpectedFiles } = require("./generate")
@@ -85,7 +86,7 @@ function main() {
     // Tnx https://blog.apollographql.com/three-ways-to-represent-your-graphql-schema-a41f4175100d!
     const text = fs.readFileSync(input, "utf8")
     const schema = graphql.buildSchema(text)
-    const res = graphql.graphqlSync(schema, graphql.introspectionQuery)
+    const res = graphql.graphqlSync({schema, source: getIntrospectionQuery()})
     if (res.data) json = res.data
     else {
       console.error("graphql parse error:\n\n" + JSON.stringify(res, null, 2))

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "mst-gql",
-  "version": "0.15.0",
+  "version": "0.15.0-graphql-upgrade",
   "description": "Bindings for mobx-state-tree and GraphQL",
   "author": "Michel Weststrate",
   "keywords": [
@@ -48,8 +48,8 @@
     "camelcase": "^6.2.0",
     "cosmiconfig": "^7.0.0",
     "fast-json-stable-stringify": "^2.1.0",
-    "graphql": "14.6.0",
-    "graphql-request": "^4.0.0",
+    "graphql": "^16.3.0",
+    "graphql-request": "^4.2.0",
     "lodash": "^4.17.21",
     "pluralize": "^8.0.0",
     "throttle-debounce": "^3.0.1"
@@ -90,6 +90,6 @@
     "mobx-state-tree": "^5.0.1"
   },
   "resolutions": {
-    "graphql": "14.6.0"
+    "graphql": "^16.3.0"
   }
 }

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "mst-gql",
-  "version": "0.15.0-graphql-upgrade",
+  "version": "0.15.0-361-graphql-upgrade",
   "description": "Bindings for mobx-state-tree and GraphQL",
   "author": "Michel Weststrate",
   "keywords": [
@@ -31,7 +31,7 @@
     "generator"
   ],
   "scripts": {
-    "test": "jest test && cd examples/2-scaffolding && yarn && yarn start && cd ../6-scaffolding-ts-hasura && yarn && yarn start ",
+    "test": "jest test && cd examples/2-scaffolding && yarn && yarn start && cd ../6-scaffolding-ts-hasura && yarn && yarn start && cd ../../tests && ./test-cra.sh && rm -rf cra-test",
     "test:debug": "node --inspect-brk node_modules/.bin/jest --runInBand",
     "watch": "jest test --watch",
     "build": "microbundle --no-compress --external mobx,mobx-react,mobx-state-tree,graphql-tag,graphql,react,react-dom,graphql-request",

--- a/tests/generator/generate.test.js
+++ b/tests/generator/generate.test.js
@@ -102,7 +102,7 @@ type query_root {
   expect(
     hasFileContent(
       findFile(output, "RootStore.base"),
-      `.extend(configureStoreMixin([['query_root', () => QueryRootModel], ['my_user', () => MyUserModel], ['possibly_empty_box', () => PossiblyEmptyBoxModel]], ['my_user', 'possibly_empty_box'], "js"))`
+      `.extend(configureStoreMixin([['my_user', () => MyUserModel], ['possibly_empty_box', () => PossiblyEmptyBoxModel], ['query_root', () => QueryRootModel]], ['my_user', 'possibly_empty_box'], "js"))`
     )
   ).toBeTruthy()
 })

--- a/tests/generator/generate.test.js
+++ b/tests/generator/generate.test.js
@@ -176,6 +176,7 @@ type Query {
       namingConvention: "asis"
     }
   )
+  console.log("+++ output", output)
   expect(output).toMatchSnapshot()
 
   expect(findFile(output, "SearchItemModelSelector")).toBeTruthy()

--- a/tests/lib/abstractTypes/models/RootStore.base.js
+++ b/tests/lib/abstractTypes/models/RootStore.base.js
@@ -3,18 +3,18 @@
 import { types } from "mobx-state-tree"
 import { MSTGQLStore, configureStoreMixin } from "mst-gql"
 
-import { SearchResultModel } from "./SearchResultModel"
-import { searchResultModelPrimitives, SearchResultModelSelector } from "./SearchResultModel.base"
 import { MovieModel } from "./MovieModel"
 import { movieModelPrimitives, MovieModelSelector } from "./MovieModel.base"
 import { BookModel } from "./BookModel"
 import { bookModelPrimitives, BookModelSelector } from "./BookModel.base"
-import { RepoModel } from "./RepoModel"
-import { repoModelPrimitives, RepoModelSelector } from "./RepoModel.base"
+import { SearchResultModel } from "./SearchResultModel"
+import { searchResultModelPrimitives, SearchResultModelSelector } from "./SearchResultModel.base"
 import { UserModel } from "./UserModel"
 import { userModelPrimitives, UserModelSelector } from "./UserModel.base"
 import { OrganizationModel } from "./OrganizationModel"
 import { organizationModelPrimitives, OrganizationModelSelector } from "./OrganizationModel.base"
+import { RepoModel } from "./RepoModel"
+import { repoModelPrimitives, RepoModelSelector } from "./RepoModel.base"
 
 import { searchItemModelPrimitives, SearchItemModelSelector  } from "./"
 import { ownerModelPrimitives, OwnerModelSelector  } from "./"
@@ -29,7 +29,7 @@ import { ownerModelPrimitives, OwnerModelSelector  } from "./"
 */
 export const RootStoreBase = MSTGQLStore
   .named("RootStore")
-  .extend(configureStoreMixin([['SearchResult', () => SearchResultModel], ['Movie', () => MovieModel], ['Book', () => BookModel], ['Repo', () => RepoModel], ['User', () => UserModel], ['Organization', () => OrganizationModel]], ['SearchResult', 'Repo']))
+  .extend(configureStoreMixin([['Movie', () => MovieModel], ['Book', () => BookModel], ['SearchResult', () => SearchResultModel], ['User', () => UserModel], ['Organization', () => OrganizationModel], ['Repo', () => RepoModel]], ['SearchResult', 'Repo']))
   .props({
     searchresults: types.optional(types.map(types.late(() => SearchResultModel)), {}),
     repos: types.optional(types.map(types.late(() => RepoModel)), {})

--- a/tests/lib/abstractTypes/models/index.js
+++ b/tests/lib/abstractTypes/models/index.js
@@ -1,13 +1,13 @@
 /* This is a mst-gql generated file, don't modify it manually */
 /* eslint-disable */
 
-export * from "./SearchResultModel"
-export * from "./SearchItemModelSelector"
 export * from "./MovieModel"
 export * from "./BookModel"
-export * from "./RepoModel"
+export * from "./SearchItemModelSelector"
+export * from "./SearchResultModel"
 export * from "./OwnerModelSelector"
 export * from "./UserModel"
 export * from "./OrganizationModel"
+export * from "./RepoModel"
 export * from "./RootStore"
 export * from "./reactUtils"

--- a/tests/lib/unionTypes/models/RootStore.base.js
+++ b/tests/lib/unionTypes/models/RootStore.base.js
@@ -3,12 +3,12 @@
 import { types } from "mobx-state-tree"
 import { MSTGQLStore, configureStoreMixin } from "mst-gql"
 
-import { TodoListModel } from "./TodoListModel"
-import { todoListModelPrimitives, TodoListModelSelector } from "./TodoListModel.base"
 import { BasicTodoModel } from "./BasicTodoModel"
 import { basicTodoModelPrimitives, BasicTodoModelSelector } from "./BasicTodoModel.base"
 import { FancyTodoModel } from "./FancyTodoModel"
 import { fancyTodoModelPrimitives, FancyTodoModelSelector } from "./FancyTodoModel.base"
+import { TodoListModel } from "./TodoListModel"
+import { todoListModelPrimitives, TodoListModelSelector } from "./TodoListModel.base"
 
 import { todoModelPrimitives, TodoModelSelector  } from "./"
 
@@ -22,7 +22,7 @@ import { todoModelPrimitives, TodoModelSelector  } from "./"
 */
 export const RootStoreBase = MSTGQLStore
   .named("RootStore")
-  .extend(configureStoreMixin([['TodoList', () => TodoListModel], ['BasicTodo', () => BasicTodoModel], ['FancyTodo', () => FancyTodoModel]], ['TodoList', 'BasicTodo', 'FancyTodo'], "js"))
+  .extend(configureStoreMixin([['BasicTodo', () => BasicTodoModel], ['FancyTodo', () => FancyTodoModel], ['TodoList', () => TodoListModel]], ['TodoList', 'BasicTodo', 'FancyTodo'], "js"))
   .props({
     todoLists: types.optional(types.map(types.late(() => TodoListModel)), {}),
     basicTodos: types.optional(types.map(types.late(() => BasicTodoModel)), {}),

--- a/tests/lib/unionTypes/models/index.js
+++ b/tests/lib/unionTypes/models/index.js
@@ -1,9 +1,9 @@
 /* This is a mst-gql generated file, don't modify it manually */
 /* eslint-disable */
 
-export * from "./TodoListModel"
 export * from "./TodoModelSelector"
 export * from "./BasicTodoModel"
 export * from "./FancyTodoModel"
+export * from "./TodoListModel"
 export * from "./RootStore"
 export * from "./reactUtils"

--- a/tests/test-cra.sh
+++ b/tests/test-cra.sh
@@ -1,0 +1,69 @@
+#!/usr/bin/env bash
+#
+# Creates a new CRA app with the latest version of mst-gql dependencies added to it. An mst model is generated using
+# the graphql schema copied from examples/6-scaffolding-ts-hasura/schema.graphql and then the app is run to
+# see if it can run all right.
+#
+#
+# Note: the generated cra-test directory should not be checked into git. Unfortunately it cannot be added to .gitignore
+# otherwise relative-deps won't work (it ignores files in .gitignore).
+#
+
+rm -rf cra-test
+
+mkdir cra-test
+cd cra-test
+
+npx create-react-app cra-app --template typescript
+cd cra-app
+cp ../../../examples/6-scaffolding-ts-hasura/schema.graphql .
+yarn add mobx mobx-state-tree mobx-react react react-dom mst-gql graphql-request
+yarn add graphql graphql-tag
+yarn add relative-deps
+
+# Add current ms-gql version as relative dependency
+# perl -0777 -pi.original -e 's|\}$|},\n  "relativeDependencies": {\n    "mst-gql": "../../../"\n  }|sm' package.json
+npx relative-deps add ../../../
+
+# Load mst-gql relative deps
+node_modules/.bin/relative-deps
+
+# Add some code to import the generated files
+perl -0777 -pi.original -e 's|(import.*?./App.css.*?;)|\1\nimport {RootStore} from "./model/RootStore";\nconst rootStore = RootStore.create({})|sm' src/App.tsx
+
+# now generate models
+npx mst-gql --format ts --outDir src/model schema.graphql
+# Compile it
+npx tsc --extendedDiagnostics
+
+# run the project and check if runs all right. Not sure how portable or exact this is, but seems to work
+rm -f out
+yarn start | tee ./out 2>&1 &
+
+echo -n "Checking if app runs all right "
+for i in {1..20}
+do
+  # Failed to compile.
+  failed=`egrep 'Failed to compile|ERROR in' out`
+  if [[ ! -z ${failed} ]]; then
+    echo
+    echo "App failed to compile"
+    pid=`ps -ef | grep tests/cra-test/cra-app | grep -v grep | awk '{print $2}' -`
+    kill -9 $pid > /dev/null 2>&1
+    exit
+  fi
+
+  no_issues=`grep 'No issues found.' out | wc -l`
+  if [ "$no_issues" -eq "1" ]; then
+    echo
+    echo "App ran successfully!"
+    pid=`ps -ef | grep tests/cra-test/cra-app | grep -v grep | awk '{print $2}' -`
+    kill -9 $pid > /dev/null 2>&1
+    exit
+  fi
+
+  echo -n "."
+
+  sleep 3
+done
+

